app/routes.py

```diff
@@ -377,31 +378,46 @@ def remove_candidate():
 def missing_route():
     return render_template("404.html")
 
-
-# the update status function is the function to be ran repaeatedly hence the id
-# the function will be ran in 30 seconds intervals
-@scheduler.task('interval', id='do_update_status', seconds=10, misfire_grace_time=900)
-def update_status():
-    now = datetime.now()
+def schedule_job(id):
+    election = Election.query.get(id)
+    # if election status is pending set to started
+    if election.status_id == 1:
+        election.status_id = 2
+    # if election status is started set to ended
+    elif election.status_id == 2:
+        election.status_id = 3
+    try:
+        db.session.commit()
+        # just to confirm new status
+        print(election.name, "has become: ", election.status.name)
+    except Exception as e:
+        db.session.rollback()
+
+# task is to run at the beginning of everyday
+@scheduler.task('cron', id='update_status_for_pending', days='*')
+def update_status_for_pending():
     app = scheduler.app
     with app.app_context():
-        elections = Election.query.all()
-        statuses = Status.query.all()
-        for election in elections:
+        # get elections to start today
+        elections = Election.query.filter(func.Date(Election.starting_at) == datetime.today().date()).all()
+        for election in elections: 
             try:
-                # check if the status of the election is pending
-                if election.status == statuses[0]:
-                    #check if the current time is greater than the starting time of the election if yes set the status of the election to started
-                    if election.starting_at < now:
-                        election.status_id = 2
-                # check if the status of the election is started
-                elif election.status == statuses[1]:
-                    #check if the current time is greater than the ending time of the election if yes set the status of the election to ended
-                    if election.ending_at < now:
-                        election.status_id = 3
+                # schedule them to run the schuled_job method at their starting datetime
+                scheduler.add_job(id="start"+election.name, func=schedule_job, trigger='date', run_date=election.starting_at, args=[election.id])    
             except Exception as e:
                 print(str(e))
+
+
+# task is to run at the beginning of everyday
+@scheduler.task('date', id='update_status_for_started', run_date=datetime.now())
```

Comment:
Why is this one `run_date=datetime.now()` and not `days="*"` like the first one for pending?

Reply:
Sorry that's a mistake its meant to be days='*'